client/List.jsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+class Item extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = {isCompleted: false};
+        this.handleClick = this.handleClick.bind(this);
+    }
+    handleClick(){
+        this.setState({isCompleted: !this.state.isCompleted});
+    }
+    render(){
+        const {name} = this.props;
+        const itemWithStatus = this.state.isCompleted ? <strike>{name}</strike> : name; 
+        return (
+        <span onClick={this.handleClick}>
+            {itemWithStatus}
+        </span>
+        );
+    }
+};
+
+const List = (props) => {
+  const { items } = props;
+
+  return (
+    <ul>{items.map((item, index) => <li key={`item_${index}`}><Item name={item}/></li>)} </ul>
+  );
+};
+
+export class DynamicList extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = { items: props.items };
+        this.handleAddItem = this.handleAddItem.bind(this);
+    }
+
+    handleAddItem(e) {
+        if (e.key == "Enter") {
+            const newItem = e.target.value.length > 0 ? e.target.value : null;
```

Comment:
Set e.target to a variable and then grab what you need from it.  React gets upset if you reference synthetic events (React's wrapped event instances) more than once.